src/ape_run/_cli.py

```diff
@@ -46,32 +47,46 @@ def _get_command(self, filepath: Path) -> Union[click.Command, click.Group, None
             logger.error_from_exception(e, f"Exception while parsing script: {relative_filepath}")
             return None  # Prevents stalling scripts
 
+        def _get_module(filepath: Path):
+            path = ".".join(
+                (str(filepath).replace(os.path.sep, ".").split("scripts.")[-1].split(".")[:-1])
+            )
+            return path
+
         # NOTE: Introspect code structure only for given patterns (do not execute it to find hooks)
         if "cli" in code.co_names:
             # If the module contains a click cli subcommand, process it and return the subcommand
             logger.debug(f"Found 'cli' command in script: {relative_filepath}")
 
             with use_temp_sys_path(filepath.parent.parent):
                 try:
-                    ns = run_module(f"scripts.{filepath.stem}")
+                    path = _get_module(filepath)
+                    ns = run_module(f"scripts.{path}")
                 except Exception as e:
                     logger.error_from_exception(
                         e, f"Exception while parsing script: {relative_filepath}"
                     )
                     return None  # Prevents stalling scripts
 
             self._namespace[filepath.stem] = ns
-            return ns["cli"]
+            cli_obj = ns["cli"]
+            cli_obj.name = filepath.stem
```

Comment:
Let's only change this if it was not already changed by the user normally.
So let's change it if it is the default `cli`:

```suggestion
            cli_obj.name = filepath.stem if cli_obj.name == "cli" else cli_obj.name
```